app/models/user.rb

```diff
@@ -36,6 +36,8 @@ class User < ApplicationRecord
   validates :reset_digest, uniqueness: true, if: :reset_digest?
   validates :activation_digest, uniqueness: true, if: :activation_digest?
 
+  after_create :generate_session_token!
```

Comment:
I think it's better to have one source of truth or a unique route to manage session tokens to reduce error surface and simplify the logic.
Managing session tokens feels as a session/controller and not a model concern.